### PR TITLE
Multiple changes in php-sdk

### PR DESCRIPTION
**Pull request by ganeshparsads**
@anantheshadiga 
@Sunil1993

### CODE REVIEW

File: src/search/SearchClient.php [hunk 1]
```diff
@@ -160,13 +187,50 @@ private function generateUrl(){
...
+							$sb.=":[".$value[$val].'+TO+'.$value[++$val].']';
+
+						if($val < $cnt)
+						{
+							$sb.= urlencode(utf8_decode(" OR ".$key));
+						}
+
+						}
+						}
+
```
Comment (Sunil1993) [inline src/search/SearchClient.php:230]: Do indentation correctly.

File: test/RecommendationsClientTest.php [hunk 2]
```diff
@@ -12,14 +12,14 @@
 class RecommendationsClientTest extends PHPUnit_Framework_TestCase{
 	
 	public function test_recommendations(){
-		Unbxd::configure("demo-u1393483043451", "ae30782589df23780a9d98502388555f", "ae30782589df23780a9d98502388555f");
+		Unbxd::configure("puravidabracelets_myshopify_com-u1405972064004", "3ad398748874973f87c1e27f7a798aa0", "3ad398748874973f87c1e27f7a798aa0");
 		$client = Unbxd::getRecommendationsClient();
```
Comment (Sunil1993) [inline test/RecommendationsClientTest.php:16]: use demosite only

File: test/SearchClientTest.php [hunk 3]
```diff
@@ -14,17 +14,18 @@
...
+		
 		$queryParams = array();
 		$queryParams["fl"]="uniqueId";
 		$queryParams["stats"] = "price";
 		$client = Unbxd::getSearchClient();
 		$response = $client->search("*",$queryParams)
-					->addFilter("color_fq",array("black"))
-					->addFilter("brand_fq",array("Ralph Lauren"))
+					->addTextFilter("color_fq",array("black"))
+					->addTextFilter("brand_fq",array("Ralph Lauren"))
```
Comment (Sunil1993) [inline test/SearchClientTest.php:28]: also add range filter and add assert accordingly

File: src/search/SearchClient.php [hunk 4]
```diff
@@ -175,8 +239,15 @@ private function generateUrl(){
 				$sb .= "&sort=".urlencode(utf8_encode(implode(",", $sorts)));
 			}
 			
+
 			$sb .= "&pageNumber=$this->pageNo";
 			$sb .= "&rows=$this->pageSize";
+			if(isset($this->otherparams) && count($this->otherparams)>0){
+				foreach ($this->otherparams as $key=>$val){
+					foreach($val as $value){
+						$sb .= "&".urlencode(utf8_encode($key.'='.$value));
```
Comment (Sunil1993) [inline src/search/SearchClient.php:248]: Do you need for loop here. cant you use $this->otherparams[$key]?

File: src/search/response/Banners.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	
+	public function getBanner(){
+		return $this->_Banner;
+	}
+	
+
+	public function getAppliedcategory(){
+		return $this->_categories;
+	}
+
```
Comment (Sunil1993) [inline src/search/response/Banners.php:28]: Where is banner.php?

File: src/recommendations/RecommendationsClient.php [hunk 6]
```diff
@@ -38,9 +38,10 @@ private function getRecommendationUrl(){
      * @return this
```
Comment (anantheshadiga) [inline src/recommendations/RecommendationsClient.php:38]: Update the comment

File: src/search/SearchClient.php [hunk 7]
```diff
@@ -54,9 +61,11 @@ private function getBrowseUrl(){
      * @return this
      */
 	
-	public function search($query/*String*/, array $queryParams/*array(String=>String)*/){
+	public function search($query/*String*/, array $queryParams=NULL){
```
Comment (anantheshadiga) [inline src/search/SearchClient.php:64]: Make the default parameter as empty array..

**Pull request closed without merging**